share_extension/ios/extension_post.js

```diff
@@ -541,23 +558,48 @@ export default class ExtensionPost extends PureComponent {
         }
     });
 
+    updateChannelsInEntities = (newChannels) => {
+        const {entities} = this.state;
+        const {channels, channelsInTeam} = entities.channels;
```

Comment:
We should be creating new objects here. I feel like this isn't going to re-render like you want it to because you're mutating the entities which won't trigger the PureComponent to re-render